sql/binlog.cc

```diff
@@ -112,13 +112,18 @@ static ulonglong binlog_global_snapshot_position;
 // Binlog position variables for SHOW STATUS
 static char binlog_snapshot_file[FN_REFLEN];
 static ulonglong binlog_snapshot_position;
+static std::string binlog_snapshot_gtid_executed;
+
+static const char *snapshot_gtid_executed_var_name = "snapshot_gtid_executed";
 
 static SHOW_VAR binlog_status_vars_detail[]=
 {
   {"snapshot_file",
    (char *)&binlog_snapshot_file, SHOW_CHAR, SHOW_SCOPE_GLOBAL},
   {"snapshot_position",
    (char *)&binlog_snapshot_position, SHOW_LONGLONG, SHOW_SCOPE_GLOBAL},
+  {snapshot_gtid_executed_var_name, &binlog_snapshot_gtid_executed[0], SHOW_CHAR,
```

Comment:
Hi @percona-ysorokin, std::string binlog_snapshot_gtid_executed was acting as the object managing allocation and deallocation of dynamic buffer holding gtid_executed string. Here I just initialized the structure with some value, which does not matter anyway (I could initialize it with nullptr), because it was always updated when variable was read by upper layer (show_binlog_vars()).
Anyway, I have changed the implementation as requested.